docker-compose.yml

```diff
@@ -1,15 +1,111 @@
-version: "3.8"
+version: '3.9'
+
+networks:
+  trainingdairy:
+    driver: bridge
 
 services:
+  nginx:
+    image: nginx:stable-alpine
+    tty: true
+    ports:
+      - "3001:80"
+    volumes:
+      - ./backend:/var/www/html
+      - ./bin/.nginx/default.conf:/etc/nginx/conf.d/default.conf
+    depends_on:
+      - php
+      - mysql
+    networks:
+      - trainingdairy
 
-  backend:
+  php:
     build:
       context: .
-      target: base
+      dockerfile: Dockerfile
+    tty: true
     environment:
-      - NODE_ENV=development
+      SERVICE_NAME: php
+      SERVICE_TAGS: dev
+      DB_HOST: ${DB_HOST}
+      DB_NAME: ${DB_NAME}
+      DB_USER: ${DB_USER}
+      DB_PASSWORD: ${DB_PASSWORD}
+      XDEBUG_MODE: develop,debug
+      XDEBUG_CONFIG:
+        client_host=host.docker.internal
+        start_with_request=yes
     volumes:
-      - ./backend:/var/www/html:rw
-      - ./frontend/dist:/var/www/html/dist/:rw
+      - ./backend:/var/www/html
+      - ./bin/.php/local.ini:/usr/local/etc/php/conf.d/local.ini
+      - ./bin/.php/conf.d/xdebug.ini:/usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini
+      - ./bin/.php/conf.d/conf.d/error_reporting.ini:/usr/local/etc/php/conf.d/error_reporting.ini
+    ports:
+      - "80:9000"
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
+    networks:
+      - trainingdairy
```

Comment:
All private data like api tokens and other secrets should never be commited to github. Instead we can use environment variables. To add one simply create `.env` file in the root (main directory, the same as this file).

## Example
1. Open `.env` file in the editor and add following
```
EXAMPLE_VARIABLE=SecretMessage
```
2. Provide the variable to `php` (backend) or `node` (frontend) to the `environment` key
```
EXAMPLE_VARIABLE: ${EXAMPLE_VARIABLE}
```
The first `EXAMPLE_VARIABLE` is the name of the variable that will be called in the service (backend/frontend). The second `EXAMPLE_VARIABLE` is the value - `SecretMessage`
3. To access the `SecretMessage` value on backend use `$_ENV("EXAMPLE_VARIABLE")`. On frontend use `import.meta.env.EXAMPLE_VARIABLE`